src/key_management/wallet_helpers.rs

```diff
@@ -94,7 +112,8 @@ pub fn generate(sig_type: SignatureType) -> Result<Vec<u8>, Error> {
             Ok(key.serialize().to_vec())
         }
         SignatureType::Delegated => {
-            unimplemented!()
+            let key = SecpPrivate::random(rng);
```

Comment:
Is it appropriate to use `SecpPrivate::random` here? If so, can you justify your choice?

Reply:
AFAIK, `Delegated` uses `Secp` private key for the `Ethereum` key and address generation (only difference being the keccak256 hash for eth addresses).

Also, I looked into the lotus `go` implementation, there [Delegated](https://github.com/filecoin-project/lotus/blob/a526c480d40898a079c806748639e8db07aa2298/lib/sigs/delegated/init.go#L19) and [Secp256k1](https://github.com/filecoin-project/lotus/blob/a526c480d40898a079c806748639e8db07aa2298/lib/sigs/secp/init.go#L18) both were using same way to generate the private key.

Am I missing something?.

Reply:
No, I think we're good.

Reply:
Were you asking because of random number value?.

Reply:
Yes, that's too.

Reply:
`OsRng` uses `getentropy(2)`, so I guess it should be safe to use, but I am no cryptography expert so do let me know, if there is a safer way.

Reply:
Shouldn't be too much of an issue.